tests/unit/lms/services/course_test.py

```diff
@@ -72,47 +75,39 @@ def test_it_inserts_False_if_theres_a_matching_row_in_course_groups_exported_fro
         ),
     )
     def test_any_with_setting(
-        self,
-        svc,
-        settings_set,
-        value,
-        expected,
-        add_courses_with_settings,
-        other_application_instance,
+        self, svc, settings_set, value, expected, add_courses_with_settings,
     ):
+        other_application_instance = factories.ApplicationInstance()
         add_courses_with_settings(settings_set)
         # Add a matching course with a different consumer key
         add_courses_with_settings(
             [{"group": {"key": value}}],
-            consumer_key=other_application_instance.consumer_key,
+            application_instance=other_application_instance,
         )
```

Comment:
I can't give a suggestion 'cos a line has gone again... but the order here reads weirdly to me here. The first create actually uses the regular `application_instance`, but a quick reading looks like it might involve the `other_application_instance`. But that's just defined ahead of time. 

I think a better ordering might be:

```python
        add_courses_with_settings(settings_set)
        
        # Add a matching course with a different consumer key
        other_application_instance = factories.ApplicationInstance()
        add_courses_with_settings(
            [{"group": {"key": value}}],
            application_instance=other_application_instance,
        )
```

I think it's good to assign this to a variable in this case, as it gives us an opportunity to name it, and reveal intent.